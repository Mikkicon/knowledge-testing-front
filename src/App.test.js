import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import {
  App,
  Landing,
  Test,
  Login,
  Register,
  TestOnLanding,
  FilterBadge
} from "./moduleExports";

describe("'Renders without crashing tests: '", () => {
  const div = document.createElement("div");

  it("renders without crashing", () => {
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`Login renders without crashing`, () => {
    ReactDOM.render(<Login />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`Register renders without crashing`, () => {
    ReactDOM.render(<Register />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`Landing renders without crashing`, () => {
    ReactDOM.render(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`FilterBadge renders without crashing`, () => {
    ReactDOM.render(<FilterBadge />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`TestOnLanding renders without crashing`, () => {
    ReactDOM.render(
      <BrowserRouter>
        <TestOnLanding />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`Test renders without crashing`, () => {
    ReactDOM.render(<Test />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe("Snapshot tests (toMatchSnapshot()):", () => {
  it(`<Test/>`, () => {
    const tree = renderer.create(<Test />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`Empty test as prop to <Test/>:`, () => {
    const tree = renderer.create(<Test testData={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`Non-Object (3) as prop to <Test/>:`, () => {
    const tree = renderer.create(<Test testData={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`<Landing/>`, () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Landing />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
