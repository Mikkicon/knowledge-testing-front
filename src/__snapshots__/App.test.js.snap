// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests (toMatchSnapshot()): <Landing/> 1`] = `
<div
  className="landingCont"
>
  <div
    className="allTestsPreviewCont"
  >
    <input
      className="searchBar"
      placeholder="Search..."
      type="text"
    />
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
    <div
      className="testPreviewCont"
    >
      <a
        href="/test"
        onClick={[Function]}
      >
        <div
          className="testName"
        >
          JavaScript
        </div>
        <img
          alt="Loading..."
          className="testImage"
          src="JS.png"
        />
        <div
          className="testBtn"
        >
          PLAY
        </div>
      </a>
    </div>
  </div>
  <button
    className="loadMoreTests"
  >
    MORE
  </button>
  <div>
    <div
      className="filterBadge"
    >
      <div
        className="filterIcon"
      />
    </div>
    <input
      className="filterToggle"
      type="checkbox"
    />
    <div
      className="filterMenu"
    />
  </div>
  <footer />
</div>
`;

exports[`Snapshot tests (toMatchSnapshot()): <Test/> 1`] = `
<div>
  <h2>
    
  </h2>
  <div
    className="container"
  >
    <div
      className="question"
    >
      1
      . 
    </div>
    <div
      className="answers"
    >
      <div />
    </div>
    <div
      className="footer"
    >
      <div
        className="pages"
      >
        <div />
      </div>
      <span
        className="divider"
      />
      <div
        className="time"
      >
        <span
          className="clock"
        />
        <span
          id="time"
        >
          0:00
        </span>
      </div>
      <span
        className="divider"
      />
      <div
        className="navigate"
      >
        <span
          onClick={[Function]}
        >
          <i
            className="arrow left"
          />
        </span>
        <span
          onClick={[Function]}
        >
          <i
            className="arrow right"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot tests (toMatchSnapshot()): Empty test as prop to <Test/>: 1`] = `
<div>
  <h2>
    
  </h2>
  <div
    className="container"
  >
    <div
      className="question"
    >
      1
      . 
    </div>
    <div
      className="answers"
    >
      <div />
    </div>
    <div
      className="footer"
    >
      <div
        className="pages"
      >
        <div />
      </div>
      <span
        className="divider"
      />
      <div
        className="time"
      >
        <span
          className="clock"
        />
        <span
          id="time"
        >
          0:00
        </span>
      </div>
      <span
        className="divider"
      />
      <div
        className="navigate"
      >
        <span
          onClick={[Function]}
        >
          <i
            className="arrow left"
          />
        </span>
        <span
          onClick={[Function]}
        >
          <i
            className="arrow right"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot tests (toMatchSnapshot()): Non-Object (3) as prop to <Test/>: 1`] = `
<div>
  <h2>
    
  </h2>
  <div
    className="container"
  >
    <div
      className="question"
    >
      1
      . 
    </div>
    <div
      className="answers"
    >
      <div />
    </div>
    <div
      className="footer"
    >
      <div
        className="pages"
      >
        <div />
      </div>
      <span
        className="divider"
      />
      <div
        className="time"
      >
        <span
          className="clock"
        />
        <span
          id="time"
        >
          0:00
        </span>
      </div>
      <span
        className="divider"
      />
      <div
        className="navigate"
      >
        <span
          onClick={[Function]}
        >
          <i
            className="arrow left"
          />
        </span>
        <span
          onClick={[Function]}
        >
          <i
            className="arrow right"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
